app/backend/src/couchers/interceptors.py

```diff
@@ -36,16 +37,19 @@ def intercept_service(self, continuation, handler_call_details):
         if not authorization.startswith("Bearer "):
             return unauthenticated_handler()
 
-        user_id = self._get_user_for_session_token(token=authorization[7:])
+        user_session = self._get_session_for_token(token=authorization[7:])
 
-        if not user_id:
+        if not user_session:
             return unauthenticated_handler()
 
+        if not self._allow_jailed and user_session.jailed:
+            return unauthenticated_handler("Jailed")
```

Comment:
Maybe don't leak the word Jail to the end user. (some clients may show the error reason straight up)

Reply:
We should pick some appropriate word.